bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -340,42 +467,10 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
         self.inner.unmark_used(&(keychain.clone(), index))
     }
 
-    /// Convenience method to call [`Self::pad_keychain_with_unused`] on all keychains.
-    ///
-    /// Returns whether any new scripts were derived.
-    pub fn pad_all_with_unused(&mut self, pad_len: u32) -> DerivationAdditions<K> {
-        let keychains = self.keychains.clone().into_keys();
-
-        let mut additions = DerivationAdditions::default();
-        for keychain in keychains {
-            additions.append(self.pad_keychain_with_unused(&keychain, pad_len));
-        }
-        additions
-    }
-
-    /// Derives and stores `pad_len` new script pubkeys after the last active index of `keychain`.
-    ///
-    /// This is useful when scanning blockchain data for transaction outputs belonging to the
-    /// keychain since the derivation index of new transactions is likely to be higher than the
-    /// current last active index.
-    pub fn pad_keychain_with_unused(
-        &mut self,
-        keychain: &K,
-        pad_len: u32,
-    ) -> DerivationAdditions<K> {
-        let up_to = self
-            .last_active_index(keychain)
-            .map(|i| i.saturating_add(pad_len))
-            .or(pad_len.checked_sub(1));
-        match up_to {
-            Some(up_to) => self.store_up_to(keychain, up_to),
-            None => Default::default(),
-        }
-    }
-
     /// Iterates over all unused script pubkeys for a `keychain` that have been stored in the index.
-    pub fn keychain_unused(&self, keychain: &K) -> impl DoubleEndedIterator<Item = (u32, &Script)> {
-        let range = (keychain.clone(), u32::MIN)..(keychain.clone(), u32::MAX);
+    pub fn unused_spks(&self, keychain: &K) -> impl DoubleEndedIterator<Item = (u32, &Script)> {
```

Comment:
This new name is worse because it is inconsistent. The `inner` which we deref to has a method name `unused` which returns unused spks and now we have a method that is named `unused_spks` which actually gets unused by keychain. `unused_for_keychain` might be better.

Reply:
Fixed in d2b0d214e6c7819aa904237c967c46bdaf66ab5e